thorlcr/thorutil/thmem.hpp

```diff
@@ -396,17 +409,49 @@ interface IWritePosCallback : extends IInterface
 
 class graph_decl CThorSpillableRowArray : private CThorExpandingRowArray, implements IThorArrayLock
 {
-    const size32_t commitDelta;  // How many rows need to be written before they are added to the committed region?
+    size32_t commitDelta;  // How many rows need to be written before they are added to the committed region?
     rowidx_t firstRow; // Only rows firstRow..numRows are considered initialized.  Only read/write within cs.
     rowidx_t commitRows;  // can only be updated by writing thread within a critical section
     mutable CriticalSection cs;
     ICopyArrayOf<IWritePosCallback> writeCallbacks;
+    CriticalSection shrinkingCrit;
+    atomic_t resizing;
+    enum ResizeState { resize_nop, resize_shrinking, resize_resizing };
+
+    class CToggleResizingState
+    {
+        ResizeState state;
+        atomic_t &resizing;
```

Comment:
Any new code should really be using std::atomic instead of atomic_t.  We should be aiming to remove all uses of atomic_t from the system.